src/utils/loadFileHandler.js

```diff
@@ -0,0 +1,130 @@
+ *    such product, must reproduce the above copyright notice, this list of
+ *    conditions and the following disclaimer in the documentation and/or other
+ *    materials provided with the distribution.
+ *
+ * 2. Neither the name of Nordic Semiconductor ASA nor the names of its
+ *    contributors may be used to endorse or promote products derived from this
+ *    software without specific prior written permission.
+ *
+ * 3. This software, with or without modification, must only be used with a Nordic
+ *    Semiconductor ASA integrated circuit.
+ *
+ * 4. Any software provided in binary form under this license must not be reverse
+ *    engineered, decompiled, modified and/or disassembled.
+ *
+ * THIS SOFTWARE IS PROVIDED BY NORDIC SEMICONDUCTOR ASA "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+ * MERCHANTABILITY, NONINFRINGEMENT, AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL NORDIC SEMICONDUCTOR ASA OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR
+ * TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
+ * THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import fs from 'fs';
+import { deserialize } from 'bson';
+import { createInflateRaw } from 'zlib';
+import { Writable } from 'stream';
+
+const setupBuffer = async filename => {
+    const buffer = Buffer.alloc(370 * 1e6);
+    let size = 0;
+    const content = new Writable({
+        write(chunk, _encoding, callback) {
+            chunk.copy(buffer, size, 0);
+            size += chunk.length;
+            callback();
+        },
+    });
+
+    await new Promise((resolve, reject) => {
+        const readStream = fs.createReadStream(filename).on('error', err => {
+            reject(err);
+        });
+        const pipedStream = readStream.pipe(createInflateRaw());
+        pipedStream.on('error', err => {
+            reject(err);
+        });
+        pipedStream.pipe(content).on('finish', resolve);
+    });
+
+    buffer.slice(0, size);
+
+    let pos = 0;
+
+    return {
+        readInitialChunk() {
+            const len = buffer.slice(pos, pos + 4).readUInt32LE();
+            const chunk = buffer.slice(pos, pos + len);
+            pos += len;
+
+            return chunk;
+        },
+        readChunk() {
+            const len = buffer.slice(pos, pos + 4).readUInt32LE();
+            pos += 4;
+            const chunk = buffer.slice(pos, pos + len);
+            pos += len;
+
+            return chunk;
+        },
+        isDepleted() {
+            pos < buffer.length;
+        },
```

Comment:
This condition seems to be just the opposite what you'd expect.

Reply:
Good catch, this is definitely a bug given that it doesn't even return anything.